### PR TITLE
Updated all files, added functions in folder helpers: setConverterVal…

### PR DESCRIPTION
**Pull request by MathewWith**
…ue, reverseCurrencies. Added folders: shared in src and api in service

### CODE REVIEW

File: src/service/api/get-all-currencies.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios'
```
Comment (CodingRuslan) [inline src/service/api/get-all-currencies.ts]: If  you will have app with 50 enpoints, will you created 50 files ?

File: src/App.tsx [hunk 2]
```diff
@@ -16,17 +16,19 @@ function App() {
       const arrayCurrencies = await getAllCurrencies()
```
Comment (CodingRuslan) [inline src/App.tsx]: array, apply clean code

File: src/App.tsx [hunk 3]
```diff
@@ -16,17 +16,19 @@ function App() {
...
       setAllCurrencies(arrayCurrencies)
     }
+    console.log('help');
```
Comment (CodingRuslan) [inline src/App.tsx]: remove console logs

File: src/App.tsx [hunk 4]
```diff
@@ -16,17 +16,19 @@ function App() {
...
+    
     getArrayCurrencies()
   }, [])
 
   return (
     <div className="App">
       <h1>Change money</h1>
 
-      {ListCurrencies(allCurrencies)}
+      {ListCurrencies(allCurrencies, leftInputValue, rightInputValue, isReversed, setLeftInputValue, setRightInputValue)}
```
Comment (CodingRuslan) [inline src/App.tsx]: Read about React components and how it's works

File: src/components/List-currencies/List-currencies.tsx [hunk 5]
```diff
@@ -1,10 +1,24 @@
-import {useState} from "react"
+import {useEffect, useState} from "react"
+import { reverseCurrencies } from "src/helpers/reverse-currency";
+import { setConvertedValue } from "src/helpers/set-converted-value";
 import { Select } from "../Select/Select"
 
-export const ListCurrencies: any = ({allCurrencies}: {allCurrencies: string[]}) => {
+export const ListCurrencies: any = (props: any) => {
```
Comment (CodingRuslan) [inline src/components/List-currencies/List-currencies.tsx]: Nice types

File: src/components/List-currencies/List-currencies.tsx [hunk 6]
```diff
@@ -1,10 +1,24 @@
...
+    const {allCurrencies, leftInputValue, rightInputValue, isReversed, setLeftInputValue, setRightInputValue} = props;
+    console.log(allCurrencies);
```
Comment (CodingRuslan) [inline src/components/List-currencies/List-currencies.tsx]: Remove console log

File: src/components/List-currencies/List-currencies.tsx [hunk 7]
```diff
@@ -1,10 +1,24 @@
...
+    
+    useEffect(() => {
```
Comment (CodingRuslan) [inline src/components/List-currencies/List-currencies.tsx]: In list of currencies you choose corrency for left and right inputs. why you added other logic in this pure components ? Why list of currencies controls reversing and so on?

File: src/components/List-currencies/List-currencies.tsx [hunk 8]
```diff
@@ -1,10 +1,24 @@
...
+        if(!isReversed) {
+            reverseCurrencies(currencyLeft, currencyRight, setCurrencyLeft, setCurrencyRight)
+        }
+        
+        isReversed 
+        ? setConvertedValue(currencyLeft, currencyRight, leftInputValue, setLeftInputValue)
```
Comment (CodingRuslan) [inline src/components/List-currencies/List-currencies.tsx]: Stypid logic. You should send isReversed variable into function and inside this function should store reversing logic

File: src/helpers/reverse-currency.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+export const reverseCurrencies = (currencyLeft: string, currencyRight: string, setCurrencyLeft: any, setCurrencyRight: any ) => {
```
Comment (CodingRuslan) [inline src/helpers/reverse-currency.ts]: How it's should work ?

File: src/helpers/set-converted-value.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { getConvertedValue } from "src/service/api/get-converted-value"
+
+export const setConvertedValue = async (  currencyLeft: string, 
+                                    currencyRight: string, 
+                                    inputValue: number, 
+                                    setInputValue: any): Promise<any> => {
```
Comment (CodingRuslan) [inline src/helpers/set-converted-value.ts]: any

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 11]
```diff
@@ -1,32 +1,44 @@
...
     getArrayCurrencies()
   }, [])
-
+  
   return (
     <div className="App">
       <h1>Change money</h1>
 
-      {ListCurrencies(allCurrencies)}
+      <ListCurrencies allCurrencies={allCurrencies}
```
Comment (CodingRuslan) [inline src/App.tsx:44]: Why component who satisfy for list of currenciew get inputs data ?

File: src/App.tsx [hunk 12]
```diff
@@ -1,32 +1,44 @@
-import React, { useEffect, useState } from 'react';
+import {useState, useEffect} from 'react';
 import './App.scss';
-import { getAllCurrencies } from 'src/service/getAllCurrencies';
+import { getAllCurrencies } from 'src/service/api/requests-to-api';
 import { ListCurrencies } from 'src/components/List-currencies/List-currencies';
```
Comment (CodingRuslan) [inline src/App.tsx]: See how you name your files

File: src/App.tsx (hunk 2 above)
Comment (CodingRuslan) [reply src/App.tsx]: Why are you ignore my comment ?

File: src/service/api/requests-to-api.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios"
```
Comment (CodingRuslan) [inline src/service/api/requests-to-api.ts]: Not logical name for this file

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 14]
```diff
@@ -1,29 +1,50 @@
...
   
-  const [allCurrencies, setAllCurrencies] = useState([''])
-  const [leftInputValue, setLeftInputValue] = useState('')
-  const [rightInputValue, setRightInputValue] = useState('')
-  const [isReversed, setIsReversed] = useState(false)
+  const [allCurrencies, setAllCurrencies] = useState<string[]>([''])
+  const [leftInputValue, setLeftInputValue] = useState<string|number>('')
+  const [rightInputValue, setRightInputValue] = useState<string|number>('')
+  const [isReversed, setIsReversed] = useState<boolean>(false)
+  const [inputCurrency, setInputCurrency] = useState('')
```
Comment (CodingRuslan) [inline src/App.tsx]: What is about type ?

File: src/App.tsx [hunk 15]
```diff
@@ -1,29 +1,50 @@
...
+  const [outputCurrency, setOutputCurrency] = useState('')
```
Comment (CodingRuslan) [inline src/App.tsx]: I can't match inputCurrency/output with left and right input value. Use the same logic for naming

File: src/App.tsx (hunk 14 above)
Comment (CodingRuslan) [inline src/App.tsx]: Sets of currencyes should be in LIstCurrencies component

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

**Comment by CodingRuslan** [comment]
Fix conflicts

File: src/App.tsx (hunk 14 above)
Comment (MathewWith) [reply src/App.tsx]: But, then I'll need to pass leftCurrency and rightCurrency to App, so I created these variables in this component
I asked you about it, and you said that it's okay

File: src/components/ListCurrencies/ListCurrencies.tsx [hunk 16]
```diff
@@ -1,15 +1,18 @@
...
-
-    const [currencyLeft, setCurrencyLeft] = useState(allCurrencies[0])
-    const [currencyRight, setCurrencyRight] = useState(allCurrencies[0])
+interface IListCurrencies {
+    allCurrencies: string[],
+    setLeftCurrency: React.Dispatch<React.SetStateAction<string>>,
+    setRightCurrency: React.Dispatch<React.SetStateAction<string>>
+}
 
+export const ListCurrencies = (props: IListCurrencies ) => {
```
Comment (CodingRuslan) [inline src/components/ListCurrencies/ListCurrencies.tsx]: Do destructions in arguments

File: src/components/Select/Select.tsx [hunk 17]
```diff
@@ -1,9 +1,4 @@
-import React from "react";
-
-interface SelectProps {
-    allCurrencies: string[] ,
-    setCurrency: React.Dispatch<React.SetStateAction<string>>
-}
+import { SelectProps } from "src/types/SelectTypes";
 
 export const Select = (props: SelectProps) => {
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: Destrunctions

File: src/service/CurrencyService.ts [hunk 18]
```diff
@@ -1,7 +1,26 @@
-import axios from 'axios'
+import axios from "axios"
+import {API_KEY} from "src/shared/consts/api-key"
+
+export const getConvertedValue = async (
+            inputCurrency: string, 
+            outputCurrency: string, 
+            inputValue: string | number): Promise<number|string>  => {
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]: Why input value string or number?

File: src/service/CurrencyService.ts [hunk 19]
```diff
@@ -1,7 +1,26 @@
...
+                
+   if(inputValue === ''){
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]: Remove this condition

File: src/components/ListCurrencies/ListCurrencies.tsx [hunk 20]
```diff
@@ -1,15 +1,18 @@
-import {useState} from "react"
+import React, {useState} from "react"
 import { Select } from "../Select/Select"
 
-export const ListCurrencies = ({allCurrencies}: {allCurrencies: string[]}) => {
-
-    const [currencyLeft, setCurrencyLeft] = useState(allCurrencies[0])
-    const [currencyRight, setCurrencyRight] = useState(allCurrencies[0])
+interface IListCurrencies {
```
Comment (CodingRuslan) [inline src/components/ListCurrencies/ListCurrencies.tsx]:
```suggestion
interface ListCurrencies {
```

File: src/service/CurrencyService.ts [hunk 21]
```diff
@@ -1,7 +1,26 @@
...
+       return ''
+   }
+   
+    const response = await axios.get(
+        `https://v6.exchangerate-api.com/v6/${API_KEY}
+        /pair/${inputCurrency}/${outputCurrency}/${inputValue}`)
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]:
```suggestion
        /pair/${inputCurrency}/${outputCurrency}/${inputValue ? inputValue  : 0 }`)
```

File: src/service/CurrencyService.ts [hunk 22]
```diff
@@ -1,7 +1,26 @@
...
 
 export const getAllCurrencies = async (): Promise<string[]> => {
-    const response = await axios.get('https://openexchangerates.org/api/currencies.json')
-    let currencies = Object.keys(response.data)
-    return currencies
-}
\ No newline at end of file
+    let response = await axios.get('https://openexchangerates.org/api/currencies.json')
+    let listCurrencies = Object.keys(response.data)
+    return listCurrencies
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]:
```suggestion
    return Object.keys(response.data)
```

File: src/service/CurrencyService.ts [hunk 23]
```diff
@@ -1,7 +1,26 @@
...
+}
+
+
+///set prettier
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]: Set prettier and remove comment

File: src/types/SelectTypes.ts [hunk 24]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
```
Comment (CodingRuslan) [inline src/types/SelectTypes.ts]: Don't create a new file for each type, create the multipurpose file

File: src/types/ListCurrenciesTypes.ts [hunk 25]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+export interface ListCurrenciesTypes {
```
Comment (CodingRuslan) [inline src/types/ListCurrenciesTypes.ts]: Don't use Types ending

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/service/CurrencyService.ts (hunk 21 above)
Comment (MathewWith) [reply src/service/CurrencyService.ts]: Wow, nice

File: src/App.tsx [hunk 26]
```diff
@@ -19,17 +21,51 @@ function App() {
...
+        setRightCurrency={setRightCurrency}
+        leftCurrency={leftCurrency}
+        rightCurrency={rightCurrency}
+        />
       <div>
-          <input type="text" onClick={() => setIsReversed(prev => !prev)} onChange={(e) => setLeftInputValue(e.target.value)}/>
-          <input type="text" onClick={() => setIsReversed(prev => !prev)} onChange={(e) => setRightInputValue(e.target.value)}/>
+          <input 
+              type="text" 
+              onClick={() => setIsReversed(prev => !prev)}
```
Comment (CodingRuslan) [inline src/App.tsx]: The same question: What happens if I will click only for left input?

File: src/components/Select/Select.tsx [hunk 27]
```diff
@@ -1,19 +1,20 @@
...
+  setCurrency,
+  pickCurrency,
+}: SelectProps) => {
+    
+  return (
+    <select onChange={(e) => setCurrency(e.target.value)}>
+      {allCurrencies.map((currency: string) => {
+        return (
+          <option key={currency}>
+            {pickCurrency ? pickCurrency : currency}
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: I think we talked about pickCurrency name

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 26 above)
Comment (MathewWith) [reply src/App.tsx]: fixed it
Befor I fixed it in another branch but forgot fixed it here

File: src/components/Select/Select.tsx (hunk 27 above)
Comment (MathewWith) [reply src/components/Select/Select.tsx]: Fixed

**Pull request merged**